learning/inference.py

```diff
@@ -223,57 +178,88 @@ def main():
         vision_callback=partial(inference_func, model),
     )
 
-    for _ in range(args.max_actions):
-        _ = agent.execute_navigator_action()
-
-        if agent.is_stuck():
-            print("Agent is stuck.")
-            break
-
-    if args.ue:
-        agent.ue.close_osc()
-
-    print("Simulation complete.")
-
-    # with Communicator("127.0.0.1", ue_port=7447, py_port=7001) as ue:
-    #     print("Connected to", ue.get_project_name())
-    #     print("Saving images to", output_dir)
-    #     ue.reset()
-
-    #     previous_action = ""
-    #     for action_step in range(args.max_actions):
-    #         # Save image
-    #         image_filename = f"{output_dir}/{action_step:04}.png"
-    #         ue.save_image(image_filename)
-    #         sleep(0.5)
-
-    #         # Predict correct action
-    #         action_to_take, action_index, action_probs = model.predict(image_filename)
-    #         action_prob = action_probs[action_index]
-
-    #         # Prevent cycling actions (e.g., left followed by right)
-    #         if action_to_take == "left" and previous_action == "right":
-    #             action_prob = action_probs.argsort()[1]
-    #             action_to_take = model.dls.vocab[action_probs.argsort()[1]]
-    #         elif action_to_take == "right" and previous_action == "left":
-    #             action_prob = action_probs.argsort()[1]
-    #             action_to_take = model.dls.vocab[action_probs.argsort()[1]]
-
-    #         # set previous_action
-    #         previous_action = action_to_take
-
-    #         print(f"Moving {action_to_take} with probabilities {action_prob:.2f}")
-
-    #         # Take action
-    #         match action_to_take:
-    #             case "forward":
-    #                 ue.move_forward(args.movement_amount)
-    #             case "left":
-    #                 ue.rotate_left(args.rotation_amount)
-    #             case "right":
-    #                 ue.rotate_right(args.rotation_amount)
-    #             case _:
-    #                 raise ValueError(f"Unknown action: {action_to_take}")
+    pbar_manager = enlighten.get_manager()
```

Comment:
I think we can have 3 progress bars:

1. For trials
2. For actions
3. For progress